### PR TITLE
Feat: add hooks for distributionLimit and projectBalance

### PR DESCRIPTION
**Pull request by johnnyd-eth**
Plus upgrade to 10.15.1

### CODE REVIEW

File: src/react/hooks/juicebox/useProjectDistributionLimit.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ */
+export function useProjectDistributionLimit(): DistributionLimit {
+  const { contracts: { fundAccessConstraintsStore } } = useJBContractContext();
+  const distributionLimit = useJbFundAccessConstraintsStoreDistributionLimitOf({
+    address: fundAccessConstraintsStore.data,
+  });
+
+  return {
+    amountWei: distributionLimit?.data?.[0],
+    currency: distributionLimit?.data?.[1] as JBCurrency,
```
Comment (wraeth-eth) [inline src/react/hooks/juicebox/useProjectDistributionLimit.ts]: This is a dangerous transformation as it could be undefined

**Review by wraeth-eth — CHANGES_REQUESTED** [review]

File: src/react/hooks/juicebox/useProjectDistributionLimit.ts (hunk 1 above)
Comment (johnnyd-eth) [reply src/react/hooks/juicebox/useProjectDistributionLimit.ts]: Have updated, LMKWYT: 

Screen Shot 2023-11-09 at 1 28 29 pm

Comment (tomquirk) [reply src/react/hooks/juicebox/useProjectDistributionLimit.ts]: or `as JBCurrency | undefined` ?

File: src/react/hooks/juicebox/useProjectBalance.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ */
+export function useProjectBalance(): bigint {
+  const { contracts } = useJBContractContext();
+  const { primaryTerminalEthStore } = contracts;
+
+  const balance = useJbSingleTokenPaymentTerminalStoreBalanceOf({
+    address: primaryTerminalEthStore.data,
+  });
+
+  return balance.data ?? 0n;
```
Comment (tomquirk) [inline src/react/hooks/juicebox/useProjectBalance.ts]: suggestion: just return `balance`, so the loading and error states are returned

File: src/react/hooks/juicebox/useProjectDistributionLimit.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+export function useProjectDistributionLimit(): DistributionLimit {
+  const { contracts: { fundAccessConstraintsStore } } = useJBContractContext();
+  const distributionLimit = useJbFundAccessConstraintsStoreDistributionLimitOf({
+    address: fundAccessConstraintsStore.data,
+  });
+
+  const currency = distributionLimit?.data?.[1]
+
+  return {
+    amountWei: distributionLimit?.data?.[0],
```
Comment (tomquirk) [inline src/react/hooks/juicebox/useProjectDistributionLimit.ts]: **suggestion** return `new Ether(distributionLimit.data[0])` here if the value exists would be nice.

I would actually return a structure like this, so we retain the query data:

```jsx
return {
   data: {
       distributionLimit: ...
       distributionLimitCurrency: ...
   },
...distributionLimit
}
```

Comment (johnnyd-eth) [reply src/react/hooks/juicebox/useProjectDistributionLimit.ts]: Done

File: src/react/hooks/juicebox/useProjectBalance.ts (hunk 2 above)
Comment (johnnyd-eth) [reply src/react/hooks/juicebox/useProjectBalance.ts]: Done

**Review by wraeth-eth — APPROVED** [review]

**Pull request merged**